go/vt/sqlparser/parse_test.go

```diff
@@ -56,6 +56,14 @@ var (
 			input:  "select a,b from t",
 			output: "select a, b from t",
 		},
+		{
+			input:  "select `a`, `'b'` from t",
+			output: "select a, 'b' from t",
+		},
+		{
+			input:  `select "'ain't'", '"hello"' from t`,
+			output: `select 'ain't', "hello" from t`,
```

Comment:
I can't say that I understand this result. I see that the expected output from the `input` query is `'ain't', "hello"` when run through MySQL, but it's valid to output this as a `SELECT` as well? `select 'ain't', "hello" from t` doesn't parse in MySQL.

Reply:
The output is just a serialized form of the parse tree, which won't necessarily itself parse in all cases. That's kind of casualty of this testing method (but there's an easy way to fix it).

```
> select "'ain't'", '"hello"';
+---------+---------+
| 'ain't' | "hello" |
+---------+---------+
| 'ain't' | "hello" |
+---------+---------+
```